File: scripts/capture-screenshots.js
const { chromium } = require('playwright');

(async () => {
    const browser = await chromium.launch({ headless: true });
    const page = await browser.newPage({ viewport: { width: 1400, height: 900 } });

    const baseUrl = 'http://127.0.0.1:8765';
    const screenshotsDir = '/Users/leo/.local/share/codex-dual/screenshots';

    console.log('Navigating to Gateway Web UI...');
    await page.goto(baseUrl);
    await page.waitForTimeout(3000);

    // Take initial screenshot (Dashboard is default)
    console.log('Capturing Dashboard...');
    await page.screenshot({ path: `${screenshotsDir}/dashboard.png` });
    console.log('  ✓ Saved dashboard.png');

    // Tab names to click
    const tabs = ['Monitor', 'Requests', 'Test', 'Compare', 'API Keys', 'Config'];

    for (const tabName of tabs) {
        console.log(`Capturing ${tabName} tab...`);
        try {
            // Find and click the tab button
            const tabButton = await page.locator(`button:has-text("${tabName}")`).first();
            await tabButton.click({ timeout: 5000 });
            await page.waitForTimeout(1500);

            const filename = tabName.toLowerCase().replace(' ', '-');
            await page.screenshot({ path: `${screenshotsDir}/${filename}.png` });
            console.log(`  ✓ Saved ${filename}.png`);
        } catch (e) {
            console.log(`  ✗ Failed: ${e.message.split('\n')[0]}`);
        }
    }

    await browser.close();
    console.log('Done!');
})();

File: scripts/capture_screenshots.py
#!/usr/bin/env python3
"""Capture Web UI screenshots for README."""

from playwright.sync_api import sync_playwright
import time

SCREENSHOTS_DIR = "/Users/leo/.local/share/codex-dual/screenshots"

def capture_screenshots():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        # Use larger viewport for better screenshots
        page = browser.new_page(viewport={"width": 1400, "height": 900})

        # Navigate to Web UI
        page.goto("http://localhost:8765/")
        page.wait_for_load_state("networkidle")
        time.sleep(2)  # Wait for animations

        # 1. Dashboard screenshot
        print("Capturing Dashboard...")
        page.screenshot(path=f"{SCREENSHOTS_DIR}/dashboard.png", full_page=False)

        # 2. Requests tab
        print("Capturing Requests...")
        page.click("text=Requests")
        page.wait_for_load_state("networkidle")
        time.sleep(1)
        page.screenshot(path=f"{SCREENSHOTS_DIR}/requests.png", full_page=False)

        # 3. Click on a request to show details (if any exist)
        try:
            # Try to click on first request row
            rows = page.locator("table tbody tr")
            if rows.count() > 0:
                rows.first.click()
                time.sleep(1)
                page.screenshot(path=f"{SCREENSHOTS_DIR}/request-detail.png", full_page=False)
                # Close modal
                page.keyboard.press("Escape")
                time.sleep(0.5)
        except Exception as e:
            print(f"Could not capture request detail: {e}")

        # 4. Test tab
        print("Capturing Test Console...")
        page.click("text=Test")
        page.wait_for_load_state("networkidle")
        time.sleep(1)
        page.screenshot(path=f"{SCREENSHOTS_DIR}/test-console.png", full_page=False)

        # 5. Compare tab
        print("Capturing Compare...")
        page.click("text=Compare")
        page.wait_for_load_state("networkidle")
        time.sleep(1)
        page.screenshot(path=f"{SCREENSHOTS_DIR}/compare.png", full_page=False)

        # 6. API Keys tab
        print("Capturing API Keys...")
        page.click("text=API Keys")
        page.wait_for_load_state("networkidle")
        time.sleep(1)
        page.screenshot(path=f"{SCREENSHOTS_DIR}/api-keys.png", full_page=False)

        # 7. Config tab
        print("Capturing Config...")
        page.click("text=Config")
        page.wait_for_load_state("networkidle")
        time.sleep(1)
        page.screenshot(path=f"{SCREENSHOTS_DIR}/config.png", full_page=False)

        # 8. Light theme screenshot
        print("Capturing Light Theme...")
        page.click("text=Dashboard")
        page.wait_for_load_state("networkidle")
        time.sleep(0.5)
        # Find and click theme toggle button
        try:
            theme_btn = page.locator("button:has(i.fa-sun), button:has(i.fa-moon)").first
            theme_btn.click()
            time.sleep(1)
            page.screenshot(path=f"{SCREENSHOTS_DIR}/dashboard-light.png", full_page=False)
            # Switch back to dark
            theme_btn.click()
        except Exception as e:
            print(f"Could not capture light theme: {e}")

        browser.close()
        print(f"\nScreenshots saved to {SCREENSHOTS_DIR}/")

if __name__ == "__main__":
    capture_screenshots()
